src/projects/TodoList/index.js

```diff
@@ -37,46 +38,86 @@ class TodoList extends React.Component {
     }
   }
   changeCompleted(i) {
-    let taskListToBeAltered = this.state.todos
-    let alteredTask = taskListToBeAltered[i]
-    alteredTask.completed = !alteredTask.completed;
-    // taskListToBeAltered.splice(i, 1, alteredTask);
-    this.setState({ todos: taskListToBeAltered });
+    this.state.todos[i].completed = !this.state.todos[i].completed
+    // let taskListToBeAltered = this.state.todos
+    // let alteredTask = taskListToBeAltered[i]
+    // alteredTask.completed = !alteredTask.completed;
+    this.setState({ todos: this.state.todos });
+  }
+  showEditField(i) {
+    let alteredTask = this.state.todos[i];
+    alteredTask.isEditing = !alteredTask.isEditing;
+    alteredTask.editingTask = alteredTask.task;
+    this.setState({ todos: this.state.todos });
+  }
+  confirmEditTask(i) {
+    let alteredTask = this.state.todos[i];
+    alteredTask.task = alteredTask.editingTask;
+    alteredTask.isEditing = !alteredTask.isEditing;
+    this.setState({ todos: this.state.todos });
+  }
+  whilstEditing(i, value) {
+    let alteredTask = this.state.todos[i];
+    alteredTask.editingTask = value;
+    this.setState({ todos: this.state.todos });
   }
   render() {
     // Before the return you can do all logic you need
+    let areAnyTodosBeingEdited = this.state.todos.find(item => {
```

Comment:
This can be a `const` its value wont change once you've assigned it

Reply:
done!